metadeploy/api/views.py

```diff
@@ -207,20 +220,80 @@ def preflight(self, request, pk=None):
         if request.method == "POST":
             return self.preflight_post(request)
 
+    def scratch_org_post(self, request, pk=None):
+        devhub_enabled = settings.DEVHUB_USERNAME
+        if not devhub_enabled:
+            return Response(status=status.HTTP_501_NOT_IMPLEMENTED)
+
+        plan = self.get_object()
+        valid_plan_type = (
+            plan.supported_orgs == SUPPORTED_ORG_TYPES.Scratch
+            or plan.supported_orgs == SUPPORTED_ORG_TYPES.Both
+        )
+        if not valid_plan_type:
+            return Response(
+                {"detail": "This plan does not support creating a scratch org."},
+                status=status.HTTP_409_CONFLICT,
+            )
+
+        new_data = request.data.copy()
+        new_data["plan"] = str(plan.pk)
+        serializer = ScratchOrgJobSerializer(data=new_data)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        # Check queue status
+        # If overfull, return 503
+        queue = django_rq.get_queue("default")
+        if len(queue) > settings.MAX_QUEUE_LENGTH:
+            return Response(
+                {"detail": "Queue is overfull. Try again later."},
+                status=status.HTTP_503_SERVICE_UNAVAILABLE,
+            )
+
+        uuid = str(uuid4())
+        scratch_org = serializer.save(uuid=uuid)
+        request.session["scratch_org_id"] = uuid
+
+        serializer = ScratchOrgJobSerializer(instance=scratch_org)
+        return Response(serializer.data, status=status.HTTP_202_ACCEPTED,)
+
+    @action(detail=True, methods=["post", "get"], permission_classes=(AllowAny,))
+    def scratch_org(self, request, pk=None):
+        if request.method == "GET":
+            pass
+            # @@@ return self.scratch_org_get(request)
+        if request.method == "POST":
+            return self.scratch_org_post(request)
+
 
 class OrgViewSet(viewsets.ViewSet):
+    permission_classes = (AllowAny,)
+
     def list(self, request):
         """
         This will return data on the user's current org. It is not a
         list endpoint, but does not take a pk, so we have to implement
         it this way.
         """
-        current_job = Job.objects.filter(
-            org_id=request.user.org_id, status=Job.Status.started
-        ).first()
-        current_preflight = PreflightResult.objects.filter(
-            org_id=request.user.org_id, status=PreflightResult.Status.started
-        ).first()
+        if "scratch_org_id" in request.session:
```

Comment:
@wlonk I think we need similar logic above in `preflight_get` and `preflight_post` so that an anonymous user can fetch an existing preflight for a plan, or post a new one. Right now I get:

```
Error: ValueError at /api/plans/M1344RO/preflight/
'AnonymousUser' value must be a positive integer or a valid Hashids string.
```

Reply:
Can we talk about anonymously posting a preflight? Perhaps @davisagli can weigh in here. What concerns do we have around that? What constraints should we consider? I feel like there should be something that makes sure that you're only triggering it for a scratch org associated with the UUID you've got in your session. Is that right?

Reply:
Yes, that's right.

Reply:
Yes, my assumption was that just like starting an installation on an MD-created scratch org, you can only see or interact with Orgs/Jobs/Preflights if you have the UUID of the org in your session.